src/android/Mail.java

```diff
@@ -196,17 +204,36 @@ public void testConnection() throws MessagingException, NoSuchProviderException
         transport.close();
     }
 
-    private Session getSession() {
+    private Session getSession() throws NoSuchAlgorithmException, KeyManagementException {
         Properties props = new Properties();
         props.put("mail.smtp.host", this.host);
         props.put("mail.smtp.port", this.port);
-        if (this.encryption == 1) {
-            // SSL Authentication            
-            props.put("mail.smtp.socketFactory.port", this.port); // SSL Port
-            props.put("mail.smtp.socketFactory.class", "javax.net.ssl.SSLSocketFactory"); // SSL factory class
-        } else if (this.encryption == 2) {
-            // TLS Authentication
-            props.put("mail.smtp.starttls.enable", "true"); // Enable StartTLS
+        if (this.encryption > 0) {
+            // StartTLS is a protocol command used to inform the email server that the email client wants to 
+            // upgrade from an insecure connection to a secure one using TLS or SSL.
+            props.put("mail.smtp.starttls.enable", "true");
+            props.put("mail.smtp.socketFactory.port", this.port);
+            if (this.encryption == 1) {
+                props.put("mail.smtp.socketFactory.class", "javax.net.ssl.SSLSocketFactory"); // SSL factory class
+                props.put("mail.smtp.ssl.protocols", "SSLv3");
+            } else if (this.encryption == 2) {
+                TrustManager[] trustManager = new TrustManager[]{new X509TrustManager() {
+                    public X509Certificate[] getAcceptedIssuers() {
+                        return null;
```

Comment:
Returning `null` means that any issuer is accepted?
Would this mean that any certificate is valid? even locally created ones?
I'm no expert here but if so, seems like an easy thing to fake.